packages/plugin-core/src/features/BacklinksTreeDataProvider.ts

```diff
@@ -283,17 +110,202 @@ export default class BacklinksTreeDataProvider
+        out.push(candidateTreeItem);
+      }
     }
+    if (_.isEmpty(out)) return undefined;
+    return out;
+  };
+
+  /**
+   * Takes found references and turn them into TreeItems that could be views in the TreeView
+   * @param refs list of found references
+   * @param fsPath fsPath of current note
+   * @param parent parent backlink of these refs.
+   * @returns list of TreeItems of found references
+   */
+  private refsToBacklinkTreeItems = (
+    refs: FoundRefT[],
+    fsPath: string,
+    parent: Backlink
+  ) => {
+    return refs.map((ref) => {
+      const lineNum = ref.location.range.start.line;
+      const backlink = new Backlink(
+        ref.matchText,
+        undefined,
+        vscode.TreeItemCollapsibleState.None
+      );
+      backlink.parentBacklink = parent;
+      backlink.description = `on line ${lineNum + 1}`;
+      backlink.tooltip = ref.matchText;
+      backlink.command = {
+        command: "vscode.open",
+        arguments: [ref.location.uri, { selection: ref.location.range }],
+        title: "Open File",
+      };
+      if (ref.isCandidate) {
+        backlink.command = {
+          command: "dendron.convertCandidateLink",
+          title: "Convert Candidate Link",
+          arguments: [
+            { location: ref.location, text: path.parse(fsPath).name },
+          ],
+        };
+      }
+      return backlink;
+    });
+  };
+
+  /**
+   * Given the fsPath of current note, return the list of backlink sources as tree view items.
+   * @param fsPath fsPath of current note
+   * @returns list of the source of the backlinks as TreeItems
+   */
+  private pathsToBacklinkSourceTreeItems = async (
+    fsPath: string,
+    isLinkCandidateEnabled: boolean | undefined,
+    sortOrder: BacklinkSortOrder
+  ) => {
+    const fileName = path.parse(fsPath).name;
+    const referencesByPath = _.groupBy(
+      await findReferences(fileName, [fsPath]),
+      ({ location }) => location.uri.fsPath
+    );
+
+    let pathsSorted: string[];
+    if (sortOrder === BacklinkSortOrder.PathNames) {
+      pathsSorted = this.shallowFirstPathSort(referencesByPath);
+    } else if (sortOrder === BacklinkSortOrder.LastUpdated) {
+      pathsSorted = Object.keys(referencesByPath).sort((p1, p2) => {
+        const ref1 = referencesByPath[p1];
+        const ref2 = referencesByPath[p2];
+
+        if (
+          ref1.length === 0 ||
+          ref2.length === 0 ||
+          ref1[0].note === undefined ||
+          ref2[0].note === undefined
+        ) {
+          Logger.error({
+            msg: "Missing info for well formed backlink sort by last updated.",
+          });
+
+          return 0;
+        }
+
+        const ref2Updated = ref2[0].note.updated;
+        const ref1Updated = ref1[0].note.updated;
+
+        // We want to sort in descending order by last updated
+        return ref2Updated - ref1Updated;
+      });
+    } else {
+      Logger.error({
```

Comment:
Looks like the `sortOrder` is exhaustively checked already, so this could be replaced with `assertUnreachable(sortOrder)`.